contracts/HolographBridge.sol

```diff
@@ -154,251 +153,209 @@ contract HolographBridge is Admin, Initializable, IHolographBridge {
+      address(0),
+      0,
+      payload
+    );
+    _operator().send{value: msg.value}(gasLimit, gasPrice, toChain, msg.sender, encodedData);
   }
 
-  function erc20in(
-    uint32 fromChain,
-    address token,
-    address from,
-    address to,
-    uint256 amount,
+  function getBridgeOutRequestPayload(
+    uint32 toChain,
+    address holographableContract,
     bytes calldata data
-  ) external onlyBridge {
-    require(_registry().isHolographedContract(token), "HOLOGRAPH: not holographed");
+  ) external view returns (bytes memory samplePayload) {
     require(
-      ERC20Holograph(token).holographBridgeIn(fromChain, from, to, amount, data) ==
-        ERC20Holograph.holographBridgeIn.selector,
-      "HOLOGRAPH: bridge in failed"
+      _registry().isHolographedContract(holographableContract) || address(_factory()) == holographableContract,
+      "HOLOGRAPH: not holographed"
     );
-  }
-
-  function erc20out(
-    uint32 toChain,
-    address token,
-    address from,
-    address to,
-    uint256 amount
-  ) external payable {
-    require(_registry().isHolographedContract(token), "HOLOGRAPH: not holographed");
-    ERC20Holograph erc20 = ERC20Holograph(token);
-    require(erc20.balanceOf(from) >= amount, "HOLOGRAPH: not enough tokens");
-    (bytes4 selector, bytes memory data) = erc20.holographBridgeOut(toChain, msg.sender, from, to, amount);
-    require(selector == ERC20Holograph.holographBridgeOut.selector, "HOLOGRAPH: bridge out failed");
-    _operator().send{value: msg.value}(
-      toChain,
-      msg.sender,
-      abi.encodeWithSignature(
-        "erc20in(uint32,address,address,address,uint256,bytes)",
-        _holograph().getChainType(),
-        token,
-        from,
-        to,
-        amount,
-        data
-      )
+    (bool success, bytes memory rawResponse) = holographableContract.staticcall(
+      abi.encodeWithSelector(HolographableEnforcer.bridgeOut.selector, toChain, msg.sender, data)
     );
-  }
-
-  function deployIn(bytes calldata data) external onlyBridge {
-    (DeploymentConfig memory config, Verification memory signature, address signer) = abi.decode(
-      data,
-      (DeploymentConfig, Verification, address)
+    require(success, "HOLOGRAPH: bridge out failed");
+    (bytes4 selector, bytes memory payload) = abi.decode(rawResponse, (bytes4, bytes));
+    uint256 jobNonce;
+    assembly {
+      jobNonce := sload(_jobNonceSlot)
+    }
+    require(selector == HolographableEnforcer.bridgeOut.selector, "HOLOGRAPH: bridge out failed");
+    bytes memory encodedData = abi.encodeWithSelector(
+      HolographableEnforcer.bridgeIn.selector,
+      jobNonce + 1,
+      _holograph().getChainType(),
+      holographableContract,
+      _registry().getHToken(_holograph().getChainType()),
+      address(0),
+      0,
+      payload
     );
-    _factory().deployHolographableContract(config, signature, signer);
+    samplePayload = abi.encodePacked(encodedData, type(uint256).max, type(uint256).max);
```

Comment:
This is for front-end to be able to run hypothetical bridge request and see what the payload will be, so that they can then run it on destination chain for an exact gas limit calculation.